Added support for ambient occlusion texture

chaoticbob opened the pull request:
Added parsing for ambient occlsuion texture using `map_Po` or `map_ao`. Those were the only two variants that I found for ambient occlusion. I've ran across decent number of models that have AO textures but the corresponding MTL files have no way to specify it. 

I looked through the code, issues, and PRs and didn't find anything for ambient occlusion.  I wasn't sure if the existing ambient (`map_Ka`) is also meant for ambient occlusion. If that's the case please feel free to reject this PR.

chaoticbob committed:
Added support for ambient occlusion texture

- Added parsing for ambient occlsuion texture using map_Po or map_ao

syoyo commented:
You can use `map_Ka` for ambient occlusion texture.

https://github.com/tinyobjloader/tinyobjloader/blob/ed48fa93268ef2c3e930370fd2fa760de30aee18/tiny_obj_loader.h#L197

Some tools treats `map_Ka` as an ambient occlusion texture, e.g. https://github.com/JordanDelcros/OBJImg

So, it'd be better add comment that `map_Ka` can be used for ambient occlusion texture, instead of defining dedicated parameter for AO. Modifying PR to simply add a comment is much appreciated!

chaoticbob committed:
Revert "Added support for ambient occlusion texture"

This reverts commit af1784dc22611b61105c5c76eb69021f3c079d8f.

chaoticbob committed:
Additional details for map_Ka

- Added comment to to indicate that map_Ka can be used for both ambient
  and ambient occlusion textures.

chaoticbob commented:
Hi - thank you for the feedback. I've made the suggested changes.

syoyo commented:
@chaoticbob Thanks! Merged!

FYI, unknown/unsupported mtl parameter is stored in `unknown_parameters` and you can look up it by `GetCustomParameter`

https://github.com/tinyobjloader/tinyobjloader/blob/ed48fa93268ef2c3e930370fd2fa760de30aee18/tiny_obj_loader.h#L312
